modules/build/src/main/scala/scala/build/Build.scala

```diff
@@ -1090,13 +1090,18 @@ object Build {
           )
         else
           options
+
       val params = value(options0.scalaParams)
 
       val scopeParams =
         if (scope == Scope.Main) Nil
         else Seq(scope.name)
 
-      buildClient.setProjectParams(scopeParams ++ value(options0.projectParams))
+      val scalaCliVersionLog =
+        if options0.internal.logCliVersion then Seq(s"scala-cli ${Constants.version}") else Nil
```

Comment:
Can't use `scala-cli` here, as contextually it may be just `scala`. What you actually need is `scala.cli.ScalaCli.fullRunnerName`... which is unfortunate, since you log it in the `build` module.
You could maybe pass it via `scala.build.input.ScalaCliInvokeData`... See how it's passed around in the code elsewhere. I wonder if `scala.build.input.ScalaCliInvokeData.progName` would be enough... 

I'm thinking this would look awkward:
```
Compiling project (Scala 3.7.0 JVM (17), scala 1.7.1)
```

We should have the  full runner name, so:
```
Compiling project (Scala 3.7.0 JVM (17), Scala code runner  1.7.1)
```